lib/api/api.js

```diff
@@ -67,6 +72,17 @@ const api = {
             // Pull the source bucket and source object separated by /
             sourceBucket = source.slice(0, slashSeparator);
             sourceObject = source.slice(slashSeparator + 1);
+            sourceVersionId = query ?
+                querystring.parse(query).versionId : undefined;
+            sourceVersionId = sourceVersionId || undefined;
+            if (sourceVersionId && sourceVersionId !== 'null') {
+                try {
+                    sourceVersionId = VID.decrypt(sourceVersionId);
+                } catch (exception) {
+                    return callback(errors.InvalidArgument
+                        .customizeDescription('Invalid version id specified'));
+                }
+            }
```

Comment:
@electrachong @LaurenSpiegel - Please look at my 2 small changes in S3 and Arsenal - https://github.com/scality/S3/commit/a829fcfa3fdcc2cc580576d2cd88e9f466522404
https://github.com/scality/Arsenal/commit/d88282075161fd3d05368a0208e08a93a42e7405